src/lib.rs

```diff
@@ -17,6 +17,21 @@
     clippy::disallowed_method
 )]
 
+trait ExactlyOne<T> {
```

Comment:
could you park this in some submodule? I would like to keep the top-level mod rather clean.